doc/rst/language/spec/tuples.rst

```diff
@@ -714,6 +715,286 @@ where a comma-separated list of components is valid.
+is made because the argument ``heavy`` is a heavy tuple due to the `in`
+intent.
+
+   *Example (tuple-argument-ref-intent.chpl)*.
+
+Tuple arguments with the `ref` intent are considered to be a special form
+of light tuple where every element is passed by reference.
+
+If a tuple argument has the `ref` intent, then actual arguments are
+restricted to heavy tuples (a tuple variable or a returned tuple). Each
+individual element behaves as though it was an argument with the `ref`
+intent.
+
+   .. code-block:: chapel
+
+      proc passTupleByRef(ref tup) {
+        tup = (64, 128);
+      }
+
+      var modTup = (0, 0);
+
+      passTupleByRef(modTup);
+
+      // Should print (64, 128).
+      writeln(modTup);
+
+   .. BLOCK-test-chapeloutput
+
+      (64, 128)
+
+When a tuple argument has the default argument intent, primitive types like
+integers are copied because their default intent is `in`. In the above
+example the intent of the formal argument ``tup`` is `ref`, and so the two
+integer elements of ``modTup`` are passed by reference. Any modifications
+made to the elements of ``tup`` will be visible when ``modTup`` is written
+to standard output.
+
+A tuple argument declared with `const` intent will work similarly to one
+with a default intent, except that all the elements of the tuple are
+considered to be `const` and cannot be modified.
+
+.. _Tuple_Return_Behavior:
+
+Tuple Return Behavior
+---------------------
+
+When a tuple is returned from a function with `ref` or `const ref` return
+intent, it must refer to some form of heavy tuple that exists outside of
+the current scope. Otherwise there is a compilation error.
+  
+Both light tuples and heavy tuples can be returned by a routine. Since
+the default return intent is to return by value, a light tuple must be
+converted to a heavy tuple when it is returned from a routine.
+
+   *Example (tuple-return-behavior.chpl)*.
+
+   .. code-block:: chapel
+
+      record R { var x: int; }
+      var a: [0..0] int;
+      var i: int;
+      var r = new R(0);
+
+      updateGlobalsAndOutput(returnTuple());
+
+      proc returnTuple() {
+        return (a, i, r); // Returns a copy of a, i, and r.
+      }
+      
+      proc updateGlobalsAndOutput(tup) {
+        a[1] = 1;
+        i = 2;
+        r.x = 3;
+        writeln(tup); // Will output (0, 0, (x = 0)).
+      }
+
+   .. BLOCK-test-chapeloutput
+
+      (0, 0, (x = 0))
+
+In the above example, ``returnTuple`` returns a heavy tuple that contains
+a copy of the array ``a``, the integer ``i``, and the record ``r``.
+
+The heavy tuple returned by ``returnTuple`` is passed to the routine
+``updateGlobalsAndOutput``. It is silently converted into a light
+tuple because the formal argument ``tup`` has the default argument intent.
+
+Because the tuple passed to ``updateGlobalsAndOutput`` is heavy and contains
+no references, the assignments made to ``a``, ``i`` and ``r`` are not
+reflected in ``tup`` when it is printed to standard output.
```

Comment:
As before it would work better for me if these were comments in the code but it's a style choice and either way is OK